### ISSUE TITLE
[Strategy Market] [Switch] Elements: When to make the tails pair

### ISSUE THREAD

**Issue by vineetsk1** [vineetsk1/Switch-Strategies#20]
Original Author: Eli Fisker
Original Link: https://getsatisfaction.com/eternagame/topics/-strategy-market-switch-elements-when-to-make-the-tails-pair

Intro 

I mentioned that EternaBot didn’t always had an idea of what parts to pair up and when. Tails are one of them. 

Strategy for tail ends 

Distance to next element from both ends of the sequence: 

For 11 bases or more at each side, give +3 for making the tail ends pair up. 

Even if just one of the sides are 11 bases or more and the other is like 6 bases, I think it could pay off to pair these 6 bases. Give +2 

Give +1 for each base pair pairing up as tail bases, as long as at least minimum six bases are left over at each side, if the aptamer switching area is between tail and aptamer. Similar demands for a MS2 sequence. Leave it 6 spacer bases that it can use as a turnoff sequence. 

If there are less than 10 bases from sequence end to next element - then penalize against tail formation - increasingly. That's if the top end of the aptamer is in the switching area and in less than 10 bases distance on both sides. Similar for MS2 

However if the top end of the aptamer is supposed to be static (locked), then it can still pay off creating tail of the two tail sequence ends as long as it is somewhat like 4 base pairs long or more. But I think the smarter way of tying this particular knot will be running JL’s strategy for this last section of the strategy. (https://getsatisfaction.com/eternagam...) 

Strategy Background 

A generally good strategy - when the tail tips are not too short, is to make them pair up with each other and then make the switch happens inside of the design. Or if only one of the tails is long in one side but not the other make the long one into a static stem. 

Basically the tails are tied up into a neck segment. 

Summed up. If it is long and it dangles, make it bind somewhere. If it is not is suited for a bind elsewhere, make it bind with itself, except if it is needed to be involved in the switch. 

Labs where the tails are too short for it to be worth to tie them are SS1, Ex4 and Ex. Similar the microRNA’s labs are an exception. There the microRNA complementary dangling tail needs to be free. As one of them initiates the switch. And favorite spot of microRNA to make this happen is specifically at the end of the RNA sequence. 

Background articles 

https://getsatisfaction.com/eternagam... 

https://getsatisfaction.com/eternagam...

**Comment by vineetsk1** [comment]
Implemented with 20 lines of code.
